File: src/models/folder.rb
# frozen_string_literal: true

require_relative 'model'
require_relative '../lib/locales_helper'

class Folder < Model
  attr_reader :name, :description, :max_age,
              :alert_regex, :number_of_mails,
              :alert_mails
  attr_writer :number_of_mails

  def initialize(name, max_age, alert_regex, description = '')
    super()
    @name = name
    @max_age = max_age
    @alert_regex = alert_regex
    @description = description
    @number_of_mails = nil
    @alert_mails = []

    validate
  end

  def validate
    rules = alert_regex.is_a?(String) || max_age.is_a?(Integer)
    errors << t('error_messages.rules_not_valid', folder: @name) unless rules
  end
end

File: frontend/app/models/folder.js
import DS from 'ember-data';

export default DS.Model.extend({
  name: DS.attr('string'),
  description: DS.attr('string'),
  max_age: DS.attr('string'),
  alert_regex: DS.attr('string'),
  number_of_mails: DS.attr('number'),
  alerts: DS.attr(),

  alert_mails: DS.hasMany('mail', {async:false}),
  mailbox: DS.belongsTo('mailbox', {async:false})
});
